src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -144,6 +149,29 @@ private bool InitializeAppDomainAssemblies(bool forceLoad)
 					Scan(assembly);
 				}
 			}
+#else
+			if (assemblies.Count == 0)
+			{
+				var context = DependencyContext.Default;
```

Comment:
I agree it will break for projects that consume statically compiled dll's via extern forwards but this is solved by adding error handling. I am not sure I know what you mean when you say Castle.Core is not in the bin folder for tests. It should be because of the NuGet reference. Why is it disappearing?